### DISCUSSION TITLE
OMR working well, want to host a local mail server behind it

### DISCUSSION THREAD

**Discussion by ghost** [Ysurac/openmptcprouter#1934] [Q&A]
OMR 0.58.4 with Debian 10 VPS: it is working well to aggregate two 15Mbps DSL links and a 20Mbps 4G

I have experience with routing and firewall on OpenBSD, but not with Linux systems. Would appreciate suggestions on how to configure OMR to run a local mail server.

The VPS has two IP, one for general traffic and one for SMTP. Incoming connections on the general IP are routed to the mail server by the default configuration. Traffic on the IP that I want to use for mail is ignored.  What I am trying to achieve is for general traffic to route through the general purpose static IP, and mail traffic to route through the dedicated mail IP. It is important that outgoing mail have the source address of the dedicated mail IP.

**Comment by Ysurac** [comment]
Maybe possible, but OpenMPTCProuter is a router, it's never a good idea to use it for something else.
If you want to add a mail server, add it on VPS side by using some docker for example. Shorewall is installed by default and can help to redirect traffic from any plublic IP where you want.

**Reply by ghost** [reply]
Apologies for not being clear enough in my question. The mail server is a separate machine, which I want to be a client of OMR. What I don't know is how to control OMR's routing to fit my design.

My case is a little more complicated than usual by having two public IP at the VPS and wanting to route incoming packets based on  destination port ( smtp ) and outgoing packets based on source IP ( mail server ). The aggregation feature of OMR is valuable enough that I would be willing to put a firewall/router in front or behind or both if need be.

**Reply by Ysurac** [reply]
So in this case it's more like https://github.com/Ysurac/openmptcprouter/wiki/VPS-Multi-IPs ? And set firewall in OpenMPTCProuter to redirect the VPS public IP you want to the IP to mail server.
OpenMPTCProuter have a firewall and another firewall (Shorewall) is on VPS.

**Reply by ghost** [reply]
I am having a hard time with this, mostly due to not having a grasp of the roles of the two systems. So now hoping to get lucky with some button-pushing. The link shows a simple set of redir rules, but I am stuck at the first step of adding one that looks like omrip1_rule. The problem is that the "ss-redir for TCP" pull-down only lists interfaces hi, hi2, hi3, and hi4. They are all enabled and running with the same public IP and port number, except hi3. hi3 is enabled but not running and it lists the same public IP but the next higher port. I don't see how or where an interface is related to a public IP on the server.

**Reply by Ysurac** [reply]
All rules should be added automatically if VPS have multiple IP.
IPs are configured before VPS boot ? If not, you have to restart omr-admin : ```systemctl restart omr-admin```

**Reply by ghost** [reply]
Additional interface is configured with static IP in /etc/network/interfaces. It is a private network address shared with another VPS. ifconfig ens7 shows info for it:

ens7: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1450
        inet 10.4.96.4  netmask 255.255.240.0  broadcast 10.4.111.255
        inet6 fe80::5800:3ff:fe51:63b4  prefixlen 64  scopeid 0x20<link>
        ether 5a:00:03:51:63:b4  txqueuelen 1000  (Ethernet)
        RX packets 74  bytes 6232 (6.0 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 13  bytes 1138 (1.1 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

I assume that it is up early in network setup. I also restarted omr-admin, then reran the wizard on the local side. But still no new remote server showing up under shadowsocks-libev.

**Reply by Ysurac** [reply]
For now only multiple IPs on same interface is tested.

**Reply by Ysurac** [reply]
And for multiples IPs, only public IPs are supported for now.

**Reply by ghost** [reply]
Would it be practical to make the private IP's interface be the only path to the internet ? The idea being to use a different machine on the private network to route based on source and destination ports. In this scenario the public IP would only be the termination of the MPTCP paths.
